src/main/java/avd/inf/jdm/rentmycar/service/CarService.java

```diff
@@ -87,4 +101,26 @@ public Boolean isValidLicensePlate(String licensePlate){
         return false;
     }
 
+    public Car createCar(String licensePlate, Short yearOfManufacture, String model, ColorType colorType, int mileage, int numberOfSeats, User user){
+        log.debug("[CarService] createCar(" + licensePlate + "," + yearOfManufacture + "," + model + "," + colorType.name() + mileage + "," + numberOfSeats + ")");
+        if (licensePlate == null || licensePlate.isEmpty()) {
+            throw new IllegalArgumentException("Licenseplate must not be empty");
+        }
+
+        if (user == null) {
+            throw new IllegalArgumentException("User must not be empty");
+        }
+        // TODO Figure out a way to determine the type of car. For now an ICE car will always be created
```

Comment:
Maybe the abstract class `Car` can have a enum `fuelType` and with a `switch` statement in the `createCar` you could check the `fuelType` and then use that to determine which subclass should be used in the creation here. Because all setters are the same for each subclass, you only have to instantiate the right subclass and the already created setters will do the rest.

That enum `fuelType` on the abstract `Car` can then also be used for searching offers matching the requested `fuelType`. So it's a double bonus.

Reply:
Good thinking, but since the API (RequestBody) already contains the type, all I have to do  is add the type to the DTO.  Which is now absent. Adding the field would solve this issue.